esp/logging/loggingagent/cassandraloggingagent/cassandralogagent.cpp

```diff
@@ -0,0 +1,621 @@
+    if (userID && *userID)
+    {
+        const char* encodedPassword = cassandra->queryProp("@dbPassWord");
+        if (!encodedPassword || !*encodedPassword)
+            throw MakeStringException(-1, "Cassandra Database password required");
+
+        decrypt(dbPassword, encodedPassword);
+        dbUserID.set(userID);
+    }
+
+    //Read information about data mapping
+    Owned<IPropertyTreeIterator> iter = cfg->getElements("LogGroup");
+    ForEach(*iter)
+    {
+        StringBuffer groupName;
+        ensureInputString(iter->query().queryProp("@name"), true, groupName, -1, "LogGroup @name required");
+        Owned<CLogGroup> logGroup = new CLogGroup(groupName.str());
+        logGroup->loadMappings(iter->query());
+        logGroups.setValue(groupName.str(), logGroup);
+        if (defaultLogGroup.isEmpty())
+            defaultLogGroup.set(groupName.str());
+    }
+
+    logSourceCount = 0;
+    Owned<IPropertyTreeIterator> iter2 = cfg->getElements("LogSourceMap/LogSource");
+    ForEach(*iter2)
+    {
+        StringBuffer name, groupName, dbName;
+        ensureInputString(iter2->query().queryProp("@name"), false, name, -1, "LogSource @name required");
+        ensureInputString(iter2->query().queryProp("@maptologgroup"), true, groupName, -1, "LogSource @maptologgroup required");
+        ensureInputString(iter2->query().queryProp("@maptodb"), true, dbName, -1, "LogSource @maptodb required");
+        Owned<CLogSource> logSource = new CLogSource(name.str(), groupName.str(), dbName.str());
+        logSources.setValue(name.str(), logSource);
+        logSourceCount++;
+    }
+
+    maxTriesGTS = cfg->getPropInt("MaxTriesGTS", DefaultMaxTriesGTS);
+    loggingTransactionApp.set(cfg->hasProp("loggingTransaction") ? cfg->queryProp("loggingTransaction") : DefaultloggingTransactionApp);
+    defaultTransactionApp.set(cfg->hasProp("defaultTransaction") ? cfg->queryProp("defaultTransaction") : DefaultTransactionApp);
+    loggingTransactionCount = 0;
+
+    //Setup Cassandra
+    initKeySpace();
+    return true;
+}
+
+void CCassandraLogAgent::initKeySpace()
+{
+    //Initialize Cassandra Cluster Session
+    cassSession.setown(new CassandraClusterSession(cass_cluster_new()));
+    if (!cassSession)
+        throw MakeStringException(-1,"Unable to create cassandra cassSession session");
+
+    StringArray opts;
+    setCassandraLogAgentOption(opts, "contact_points", dbServer.str());
+    if (!dbUserID.isEmpty())
+    {
+        setCassandraLogAgentOption(opts, "user", dbUserID.str());
+        setCassandraLogAgentOption(opts, "password", dbPassword.str());
+    }
+
+    cassSession->setOptions(opts);
+
+    //prepare defaultDB
+    ensureKeySpace();
+
+    //prepare transSeed tables
+    initTransSeedTable();
+
+    //Read logging transaction seed
+    queryTransactionSeed(loggingTransactionApp.get(), transactionSeed);
+}
+
+void CCassandraLogAgent::ensureKeySpace()
+{
+    CassandraSession s(cass_session_new());
```

Comment:
It does not work. The cassSession is a CassandraClusterSession, not CassandraSession. To connect without keyspace, I have to use CassandraSession.